sycl/unittests/Extensions/CommandGraph/Update.cpp

```diff
@@ -399,3 +399,14 @@ TEST_F(WholeGraphUpdateTest, EmptyNode) {
   auto GraphExec = Graph.finalize(experimental::property::graph::updatable{});
   GraphExec.update(UpdateGraph);
 }
+
+TEST_F(CommandGraphTest, CheckFinalizeBehavior) {
+  // Check that both finalize with and without updatable property work as
+  // expected
+  auto Node = Graph.add(
+      [&](sycl::handler &cgh) { cgh.single_task<TestKernel<>>([]() {}); });
+
+  ASSERT_NO_THROW(Graph.finalize(experimental::property::graph::updatable{}));
+
+  ASSERT_NO_THROW(Graph.finalize());
```

Comment:
Can we make a throwing case too?

Reply:
I don't believe it is possible to trigger a throwing case from user code. The scheduler command to udpate a graph will query whether a command handle is required to be passed when updating a kernel, so an exception here indicates either a problem in UR or in the implementation.